src/platform/file_manager_cache.c

```diff
@@ -0,0 +1,166 @@
+#include "file_manager_cache.h"
+
+#if defined(__vita__) || defined(__SWITCH__)
+
+#include "core/string.h"
+#include "platform/file_manager.h"
+
+#include <dirent.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+
+enum {
+    STAT_DOESNT_WORK = -1,
+    STAT_UNTESTED = 0,
+    STAT_WORKS = 1
+};
+
+static dir_info *base_dir_info;
+static int stat_status;
+
+const dir_info *platform_file_manager_cache_get_dir_info(const char *dir)
+{
+    dir_info *info = base_dir_info;
+    while (info) {
+        if (strcmp(info->name, dir) == 0) {
+            return info;
+        }
+        if (!info->next) {
```

Comment:
This check is not necessary: `info = info->next` will set info to null, and the next while loop evaluation will stop the loop.

Reply:
Forgotten remnants of past code...